### ISSUE TITLE
Bug: Second reviewer not getting new responses from applicant after 2nd re-submission to LOQ

### ISSUE THREAD

**Issue by nmadruga** [openmsupply/application-manager-web-app#1196]
To reproduce this problem:

- User 1 requests clarification
- Applicant makes the change
- User 2 now picks it up and the application goes back to a new review, they don't get a re-review option

See Adam for more details (posted on Subforma on 10th May 2022)

**Comment by adamdewey** [comment]
Sorry Nicole, I think maybe my explanation wasn't clear enough (judging by the interpretation of the title!)

I was meaning the situation where there are more than one reviewer with the same role in an organisation. Both reviewers can see the same pool of applications to self-assign from .

So the first reviewer to self-assign it then sends it back to the applicant with a LOQ. The applicant re-submits it back to the pool. This time the other reviewer self-assigns it. 

My intuitive response was that the 2nd reviewer should see the original review from the 1st reviewer, and the responses from the applicant, and be able to continue from where the 1st reviewer left off (i.e. they would see the same thing as the case where the 1st reviewer draws from the pool). 

This will be especially important in bigger organisations where more than 1 person does a particular role, and also if someone goes off sick and another user has to step in while they are away. 

Hope that makes sense?

**Comment by adamdewey** [comment]
Some pics to try help explain: 

![image](https://user-images.githubusercontent.com/19318937/168046344-c6c69fa4-4281-404c-a8eb-e378441eb292.png)

![image](https://user-images.githubusercontent.com/19318937/168046359-84b3c619-cea8-41ce-8bd8-d76a551c0f8a.png)

![image](https://user-images.githubusercontent.com/19318937/168046387-a7cf6ddb-a250-4092-b2cc-8521df45938c.png)

![image](https://user-images.githubusercontent.com/19318937/168046415-7f531956-e398-4c68-a489-b1e9a658c1e2.png)

![image](https://user-images.githubusercontent.com/19318937/168046437-0a4374f2-a226-48b4-9665-bc254db68646.png)

![image](https://user-images.githubusercontent.com/19318937/168046455-59c9e605-92e4-49ed-b41f-6af4e56c37cf.png)

**Comment by adamdewey** [comment]
Ok, it's not _quite_ true that Reviewer 2 has no idea that Reviewer 1 has already looked at the application. 

If they go to the Overview tab then they can see that Reviewer 1 has taken some action 

![image](https://user-images.githubusercontent.com/19318937/168047163-2589b52d-f377-4429-a17d-be5fa3dd9af8.png)


Although clicking on the 'review' hyperlink takes them to a 404 (only Reviewer 1 can see that review apparently? Possibly something wrong with my permissions set up, although Reviewer 1 and 2 have the same permissions)

![2022-05-12_20-09-51](https://user-images.githubusercontent.com/19318937/168047339-fdf5aa84-1017-4f08-9df1-39f59f4bfec8.png)

But it's still confusing for Reviewer 2 that on the Dashboard genuinely new applications appear the same as applications that are partway through a review process by another Reviewer (i.e. grouped into the same 'available for self-assignment' filter).

**Comment by nmadruga** [comment]
> Although clicking on the 'review' hyperlink takes them to a 404 (only Reviewer 1 can see that review apparently? Possibly something wrong with my permissions set up, although Reviewer 1 and 2 have the same permissions)
> 
> ![2022-05-12_20-09-51](https://user-images.githubusercontent.com/19318937/168047339-fdf5aa84-1017-4f08-9df1-39f59f4bfec8.png)

In the system you shouldn't have permission to view another person review. The current setup of permissions (considering row level permissions) is not fully finalised yet, so a few things can happen which shouldn't, but in this case it seems appropriated. We want to have a special permission to allow (higher level reveiwers - for example who does the consolidation) to be able to view other reviews not done by them: with this issue #524

**Comment by nmadruga** [comment]
> Sorry Nicole, I think maybe my explanation wasn't clear enough (judging by the interpretation of the title!)

Yes, I completely misunderstood this. The explanation was too simple though, is almost like a philosophical issue haha

> My intuitive response was that the 2nd reviewer should see the original review from the 1st reviewer

We though about doing this, but since it would add too much complexity and also - sensitive information, you wouldn't want someone else to "continue" your work and see your review. So we have done the re-assignment as a **new review**, I thing if you would like to change that we need to discuss more. And ideally receive an external requirement - that's the rule for new features work. It's a big change, and many things should be considered (which means time to discuss too).

> An alternative solution would be to not let anyone see that the re-review is available for self-assignment apart from the original reviewer, but I don't think this is a satisfactory solution as bigger organisations may have more than 1 person that does a particular role, and also if someone goes off sick another user has to step in while they are away.

Yes, I agree that's not ideal because we add the re-assignment exactly to allow someone else to "continue" the work. But yes, in this case it's actually do a new review. I don't see the need to change this for now though.

> But it's still confusing for Reviewer 2 that on the Dashboard genuinely new applications appear the same as applications that are partway through a review process by another Reviewer (i.e. grouped into the same 'available for self-assignment' filter).

Hmmmm there is a way to differ this saying in particular. We haven't used this yet, but we used to have the option to set the new submission by the applicant as RE-SUBMITTED and therefor we could have a different filter to display, something like "available for taking-over assignment after new submission" or something shorter than that. Would that be a reasonable solution for now?

**Comment by adamdewey** [comment]
>We though about doing this, but since it would add too much complexity and also - sensitive information, you wouldn't want someone else to "continue" your work and see your review. 

Oh this is surprising to me - I didn't think it would be sensitive information to share a review between internal staff? 

I guess that must depend on the organisation - it seems in PNG that anyone working in the same area could pick up where another person left off. 

Maybe we need to cross that bridge when we get to the exact requirements gathering for PNG. 

>Hmmmm there is a way to differ this saying in particular. We haven't used this yet, but we used to have the option to set the new submission by the applicant as RE-SUBMITTED 

I think this might be ok, but also I feel like there is a type of status missing? 

We have the decision that we don't want internal staff to view other internal staff's reviews and that may be fair enough, but in the real world if you had that application on paper then you would probably keep it in your desk or office and so the 2nd Reviewer wouldn't see that it's available to start working on anyway. 

If we only want 1 Reviewer to work on an application at one time then I think we need a `Checked out` or `Being worked on` status for applications so that it doesn't distract other Reviewers who might be looking at the same pool of work?

**Comment by adamdewey** [comment]
I guess if it was set to `RE-SUBMITTED` you could configure it so that this shouldn't appear in the dashboard filters except for the original Reviewer. 

I.e. infer that anything other than `SUBMITTED` means it's being worked on by someone else?

**Comment by nmadruga** [comment]
> If we only want 1 Reviewer to work on an application at one time then I think we need a `Checked out` or `Being worked on` status for applications so that it doesn't distract other Reviewers who might be looking at the same pool of work?

Well we do that now: after it has been re-assigned the previous reviewer can no longer start or submitted (in case already started) the review. There are problems with that which should be address during some of the work requested in: #1214

**Comment by nmadruga** [comment]
> I guess if it was set to `RE-SUBMITTED` you could configure it so that this shouldn't appear in the dashboard filters except for the original Reviewer.
> 
> I.e. infer that anything other than `SUBMITTED` means it's being worked on by someone else?

For now we won't be changing this since it isn't a formal requirement yet. But it is possible to change that, which is what I was explaining because in our initial plan we though about sending a different event when the user re-submitted (not done yet though, so we can't change the filters - which would be a configuration only)

**Comment by adamdewey** [comment]
>Well we do that now: after it has been re-assigned the previous reviewer can no longer start or submitted (in case already started) the review. There are problems with that which should be address during some of the work requested in: #1214 

Hmm, I think I'm still missing something. The way the PNG system is currently set up doesn't feel right, but maybe it's something I haven't set up correctly in configuration. 

Perhaps it's possible via config to automatically assign the re-submit back to the original reviewer so the other reviewers wouldn't ever see it as outstanding in their dashboard filters? 

And then the original reviewer could un-assign it from themselves if they no longer want to work on it (and therefore make it available to the other reviewers)?

**Comment by adamdewey** [comment]
Or perhaps I'm just thinking of the intended use of the dashboard incorrectly, perhaps the user will never really look at the filters on the dashboard (if they will always be showing outstanding applications in the high numbers) and instead just look at the application list and use the filters in there to find their outstanding work?
